base/lnls-prune-artifacts.sh

```diff
@@ -77,12 +77,36 @@ get_used_epics_modules() {
     filter_out_paths "$all_modules" "$unused_modules"
 }
 
+# Traverse ancestor directories of each provided path, and concatenate all
+# their .lnls-keep-paths defined entries as absolute paths.
+get_defined_paths_to_keep() {
+    for path; do
+        depth=$(echo $path | grep -o '/' | wc -l)
+
+        for i in $(seq 1 $depth); do
+            ancestor=$(echo $path | cut -d '/' -f 1-$i)
```

Comment:
Why can't this be a `dirname` loop that terminates on `/`?

Reply:
Could also break on `.` and `/`, so it still works for local directories.

Reply:
> I like the dirname loop because it's already used elsewhere, so it's a pattern that's easy to spot.

Yeah, I agree on that. I changed this locally. I'm testing if it is still working as expected before pushing.

> Could also break on . and /, so it still works for local directories.

Indeed. I'll defer that to the future if needed. As I said, it actually is not a big deal for now.